ingestion/src/metadata/ingestion/source/database/trino.py

```diff
@@ -33,6 +40,89 @@
 from metadata.utils.logger import ingestion_logger
 
 logger = ingestion_logger()
+ROW_DATA_TYPE = "row"
+ARRAY_DATA_TYPE = "array"
+
+
+def get_type_name_and_opts(type_str: str) -> Tuple[str, Optional[str]]:
+    match = re.match(r"^(?P<type>\w+)\s*(?:\((?P<options>.*)\))?", type_str)
+    if not match:
+        util.warn(f"Could not parse type name '{type_str}'")
+        return sqltypes.NULLTYPE
+    type_name = match.group("type")
+    type_opts = match.group("options")
+    return type_name, type_opts
+
+
+def parse_array_data_type(type_str: str) -> str:
+    type_name, type_opts = get_type_name_and_opts(type_str)
+    final = type_name + "<"
+    if type_opts:
+        if type_opts.startswith(ROW_DATA_TYPE):
+            final += parse_row_data_type(type_opts)
+        elif type_opts.startswith(ARRAY_DATA_TYPE):
+            final += parse_array_data_type(type_opts)
+        else:
+            final += type_opts
+    return final + ">"
+
+
+def parse_row_data_type(type_str: str) -> str:
```

Comment:
I am getting a bit lost with the logic of these functions. Adding some docstrings with examples and a couple of unit tests would be great 🙏 These functions will be the bread and butter of Trino so it's important we can understand and evolve them safely!